server/fleet/hosts.go

```diff
@@ -250,21 +250,150 @@ type MDMHostData struct {
 	// decode an encryption key for the host.
 	EncryptionKeyAvailable bool `json:"encryption_key_available" db:"-" csv:"-"`
 
+	// this is set to nil if the key exists but decryptable is NULL in the db, 1
+	// if decryptable, 0 if non-decryptable and -1 if no disk encryption key row
+	// exists for this host. Used internally to determine the disk_encryption
+	// status and action_required fields. See MDMHostData.Scan as for where this
+	// gets filled.
+	rawDecryptable *int
+
 	// Profiles is a list of HostMDMProfiles for the host. Note that as for many
 	// other host fields, it is not filled in by all host-returning datastore methods.
 	//
 	// It is a pointer to a slice so that when set, it gets marhsaled even
 	// if the slice is empty, but when unset, it doesn't get marshaled
 	// (e.g. we don't return that information for the List Hosts endpoint).
 	Profiles *[]HostMDMAppleProfile `json:"profiles,omitempty" db:"profiles" csv:"-"`
+
+	// MacOSSettings indicates macOS-specific MDM settings for the host, such
+	// as disk encryption status and whether any user action is required to
+	// complete the disk encryption process.
+	//
+	// It is not filled in by all host-returning datastore methods.
+	MacOSSettings *MDMHostMacOSSettings `json:"macos_settings,omitempty" db:"-" csv:"-"`
+}
+
+type DiskEncryptionState string
+
+const (
+	DiskEncryptionApplied             DiskEncryptionState = "applied"
+	DiskEncryptionActionRequired      DiskEncryptionState = "action_required"
+	DiskEncryptionEnforcing           DiskEncryptionState = "enforcing"
+	DiskEncryptionFailed              DiskEncryptionState = "failed"
+	DiskEncryptionRemovingEnforcement DiskEncryptionState = "removing_enforcement"
+)
+
+func (s DiskEncryptionState) addrOf() *DiskEncryptionState {
```

Comment:
TIL of this pattern! I like it!

Reply:
Oh actually it makes sense, if it allowed this you could override a constant by using `*ptr = "new value"`. I didn't click at first...